core/include/detray/masks/single3.hpp

```diff
@@ -30,8 +31,8 @@ namespace detray {
  *
  **/
 template <unsigned int kCheckIndex, typename intersector_t = planar_intersector,
-          typename mask_links_t = __plugin::cartesian2<detray::scalar>,
-          typename mask_local_t = bool,
+          typename mask_local_t = __plugin::cartesian2<detray::scalar>,
+          typename mask_links_t = dindex,
```

Comment:
Shouldn't it have two values such as `std::tuple<dindex, dindex>` or `std::array<dindex,2>`?
Also same for other mask types

Reply:
In principle you are right, but this will become even more complicated, when the ```surface_finder``` links must resolve a type. So I would like to leave this for one of the next PRs